src/terminal/Screen_test.cpp

```diff
@@ -2911,11 +2911,34 @@ TEST_CASE("OSC.4")
 
     SECTION("query")
     {
-        screen.write("\033]4;1;?");
+        screen.write("\033]4;7;?\033\\");
         INFO(term.replyData);
+        REQUIRE(e(term.replyData) == e("\033]4;7;rgb:c0c0/c0c0/c0c0\033\\"));
     }
 
-    SECTION("set") { screen.write("\033]4;1;rgb:ab/cd/ef"); }
+    SECTION("set color via format rgb:RR/GG/BB")
+    {
+        screen.write("\033]4;7;rgb:ab/cd/ef\033\\");
+        screen.write("\033]4;7;?\033\\");
+        INFO(term.replyData);
+        REQUIRE(e(term.replyData) == e("\033]4;7;rgb:abab/cdcd/efef\033\\"));
+    }
+
+    SECTION("set color via format #RRGGBB")
+    {
+        screen.write("\033]4;7;#abcdef\033\\");
+        screen.write("\033]4;7;?\033\\");
+        INFO(term.replyData);
+        REQUIRE(e(term.replyData) == e("\033]4;7;rgb:abab/cdcd/efef\033\\"));
```

Comment:
`urxvt` (rxvt-unicode) replies with `rgb:ab00/cd00/ef00` if the color was set as `#abcdef`

Reply:
I compared that with Kitty, which was doing it like that. I should probably have to ask @dankamongmen for some feedback on if that would have any implications, if notcurses is getting the reply as `ab00/cd00/ef00` instead of `abab/cdcd/efef`. I do not know how he's mapping these replies back to 24-bit RGB. It would also break the expecations when setting via rgb: syntax and reading back the query. I think that's why it is okay (for now) to leave the PR's logic as is, until @dankamongmen objects differently. :)

Reply:
oooh let me go take a look; i hadn't considered this i don't think

Reply:
@dankamongmen i've merged to master now. but in case of anything bogus on your end, i'll hotfix ASAP. I hope it'll all be well though :-)